src/watertap_contrib/seto/unit_models/surrogate/vagmd_surrogate.py

```diff
@@ -0,0 +1,1233 @@
+
+        for k in state_dict_water.keys():
+            if state_dict_water[k].is_indexed():
+                state_args_water[k] = {}
+                for m in state_dict_water[k].keys():
+                    state_args_water[k][m] = state_dict_water[k][m].value
+            else:
+                state_args_water[k] = state_dict_water[k].value
+
+        for p, j in self.avg_condenser_props.phase_component_set:
+            if p == "Vap" and j == "H2O":
+                state_args_water["flow_mass_phase_comp"][(p, j)] = 0
+
+        init_log.info_low("Starting initialization of avg_condenser_props ")
+        self.avg_condenser_props.initialize(
+            outlvl=outlvl,
+            optarg=optarg,
+            solver=solver,
+            state_args=state_args_water,
+        )
+
+        state_args_permeate = deepcopy(state_args)
+        for p, j in self.permeate_props.phase_component_set:
+            if j == "TDS":
+                state_args_permeate["flow_mass_phase_comp"][(p, j)] = 0
+
+        init_log.info_low("Starting initialization of permeate_props ")
+        self.permeate_props.initialize(
+            outlvl=outlvl,
+            optarg=optarg,
+            solver=solver,
+            state_args=state_args_permeate,
+        )
+
+        if degrees_of_freedom(self) != 0:
+            raise Exception(
+                f"Degrees of freedom of {self.name} is {degrees_of_freedom(self)}"
+                f"before the initialization of VAGMD unit model."
+            )
+
+        opt = get_solver(solver, optarg)
+
+        with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
+            res = opt.solve(self, tee=slc.tee)
+            print_close_to_bounds(self)
+            print_infeasible_constraints(self)
+        init_log.info("Initialization status {}.".format(idaeslog.condition(res)))
+
+    """
+    Equation to calculate pressure drop
+    """
+
+    def _get_pressure_drop(self, flow_rate, salinity):
+        if self.config.module_type == "AS7C1.5L":
+            coefficients = [
+                -158.2007422,
+                0.39402609,
+                0,
+                0.000585345,
+                8.93618e-5,
+                -0.000287828,
+            ]
+        else:  # self.config.module_type == 'AS26C7.2L'
+            coefficients = [
+                -72.53793298,
+                0.110437201,
+                0,
+                0.000643495,
+                0.000189924,
+                -0.001111447,
+            ]
+
+        return (
+            coefficients[0]
+            + coefficients[1] * flow_rate
+            + coefficients[2] * salinity
+            + coefficients[3] * flow_rate * salinity
+            + coefficients[4] * flow_rate**2
+            + coefficients[5] * salinity**2
+        )
+
+    """
+    Surrogate equations for membrane performance
+    """
+
+    def _get_membrane_performance(self, TEI, FFR, TCI, SgL):
+        # Model parameters
+        PFluxAS26 = [
```

Comment:
Seems like local assignment of these parameters aren't all necessary since these are for both modules. Technically each group of parameters be created based on which module was specified, right?

Reply:
Not sure if I understood this comment. Is it trying to make the script more concise? If so, I could be wrong but I don't see a clear way to do so.